src/Microsoft.Azure.SignalR.Management/ServiceManagerBuilder.cs

```diff
@@ -39,16 +42,22 @@ public ServiceManagerBuilder WithCallingAssembly()
         /// <returns>The instance of the <see cref="IServiceManager"/>.</returns>
         public IServiceManager Build()
         {
-            _options.ValidateOptions();
-
+            _services.PostConfigure<ServiceManagerOptions>(o => o.ValidateOptions());
+            _services.AddSingleton<IConfigureOptions<ServiceManagerContext>, ServiceManagerContextSetup>();
+            _serviceProvider = _services.BuildServiceProvider();
+            var context = _serviceProvider.GetRequiredService<IOptions<ServiceManagerContext>>().Value;
             var productInfo = ProductInfo.GetProductInfo(_assembly);
-            var context = new ServiceManagerContext()
-            {
-                ProductInfo = productInfo
-            };
```

Comment:
ProductInfo is used to identify Product in UserAgent for BI use. Please help take care when switch in future version. As management SDK is also depended by function extensions.

Reply:
I'm sorry that the code about productInfo is there but I didn't upload it.